Unable to run the dashboard via docker compose

The Dockerfile for the dashboard microservice doesn't seem to be working. I am using Linux Mint and `docker compose up` gives me the following error:

```
mycryptobot-dashboard-1          | 
mycryptobot-dashboard-1          | > MyCryptoBot-dashboard@0.1.0 start /app
mycryptobot-dashboard-1          | > react-scripts start
mycryptobot-dashboard-1          | 
mycryptobot-dashboard-1          | sh: react-scripts: not found
mycryptobot-dashboard-1          | npm ERR! code ELIFECYCLE
mycryptobot-dashboard-1          | npm ERR! syscall spawn
mycryptobot-dashboard-1          | npm ERR! file sh
mycryptobot-dashboard-1          | npm ERR! errno ENOENT
mycryptobot-dashboard-1          | npm ERR! MyCryptoBot-dashboard@0.1.0 start: `react-scripts start`
mycryptobot-dashboard-1          | npm ERR! spawn ENOENT
mycryptobot-dashboard-1          | npm ERR! 
mycryptobot-dashboard-1          | npm ERR! Failed at the MyCryptoBot-dashboard@0.1.0 start script.
mycryptobot-dashboard-1          | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
mycryptobot-dashboard-1          | npm WARN Local package.json exists, but node_modules missing, did you mean to install?
mycryptobot-dashboard-1          | 
mycryptobot-dashboard-1          | npm ERR! A complete log of this run can be found in:
mycryptobot-dashboard-1          | npm ERR!     /root/.npm/_logs/2023-12-11T19_08_38_716Z-debug.log
```

If I uncomment `RUN npm install react-scripts -g` then I get a similar error saying that typescript is missing, but installing typescript doesn't seem to solve the problem:
```
mycryptobot-dashboard-1          | > MyCryptoBot-dashboard@0.1.0 start /app
mycryptobot-dashboard-1          | > react-scripts start
mycryptobot-dashboard-1          | /usr/local/lib/node_modules/react-scripts/node_modules/resolve/lib/sync.js:113
mycryptobot-dashboard-1          |     throw err;
mycryptobot-dashboard-1          |     ^
mycryptobot-dashboard-1          | 
mycryptobot-dashboard-1          | Error: Cannot find module 'typescript' from '/app/node_modules'
mycryptobot-dashboard-1          |     at Function.resolveSync [as sync] (/usr/local/lib/node_modules/react-scripts/node_modules/resolve/lib/sync.js:111:15)
mycryptobot-dashboard-1          |     at getModules (/usr/local/lib/node_modules/react-scripts/config/modules.js:119:32)
mycryptobot-dashboard-1          |     at Object.<anonymous> (/usr/local/lib/node_modules/react-scripts/config/modules.js:142:18)
mycryptobot-dashboard-1          |     at Module._compile (internal/modules/cjs/loader.js:1147:30)
mycryptobot-dashboard-1          |     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1167:10)
mycryptobot-dashboard-1          |     at Module.load (internal/modules/cjs/loader.js:996:32)
mycryptobot-dashboard-1          |     at Function.Module._load (internal/modules/cjs/loader.js:896:14)
mycryptobot-dashboard-1          |     at Module.require (internal/modules/cjs/loader.js:1036:19)
mycryptobot-dashboard-1          |     at require (internal/modules/cjs/helpers.js:72:18)
mycryptobot-dashboard-1          |     at Object.<anonymous> (/usr/local/lib/node_modules/react-scripts/config/webpack.config.js:28:17) {
mycryptobot-dashboard-1          |   code: 'MODULE_NOT_FOUND'
mycryptobot-dashboard-1          | }
mycryptobot-dashboard-1          | npm ERR! code ELIFECYCLE
mycryptobot-dashboard-1          | npm ERR! errno 1
mycryptobot-dashboard-1          | npm ERR! MyCryptoBot-dashboard@0.1.0 start: `react-scripts start`
mycryptobot-dashboard-1          | npm ERR! Exit status 1
mycryptobot-dashboard-1          | npm ERR! 
mycryptobot-dashboard-1          | npm ERR! Failed at the MyCryptoBot-dashboard@0.1.0 start script.
mycryptobot-dashboard-1          | npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
mycryptobot-dashboard-1          | npm WARN Local package.json exists, but node_modules missing, did you mean to install?
mycryptobot-dashboard-1          | 
mycryptobot-dashboard-1          | npm ERR! A complete log of this run can be found in:
mycryptobot-dashboard-1          | npm ERR!     /root/.npm/_logs/2023-12-11T19_13_43_114Z-debug.log
mycryptobot-dashboard-1 exited with code 1
```
Not quite sure where the problem lies, if you have any ideas let me know. Thanks in advance

> Apologies, I edited my comment after you saw it I guess, I meant `RUN npm install --silent`. This was what was before I made the changes, and it was working correctly. The package.json was not changed.
> 
> The dockerfile before the changes:
> 
> ```dockerfile
> FROM node:13.12.0-alpine
> 
> # set working directory
> WORKDIR /app
> 
> # add `/app/node_modules/.bin` to $PATH
> ENV PATH /app/node_modules/.bin:$PATH
> ENV DIRECTORY dashboard
> 
> # install app dependencies
> COPY $DIRECTORY/package.json $DIRECTORY/package-lock.json ./
> RUN npm install --silent
> #RUN npm install react-scripts -g
> 
> # add app
> COPY ./$DIRECTORY ./
> 
> # start app
> CMD ["npm", "start"]
> ```

- This gives the `react-scripts: not found` error. 
- Uncommenting `RUN npm install react-scripts -g
` gives the `Error: Cannot find module 'typescript' from '/app/node_modules'` error. 
- Adding `RUN npm install typescript -g` still gives a typescript not found error.

I'm rebuilding the images every time with no caching using `sudo docker compose build --no-cache && sudo docker compose up` just to be sure, but nothing seems to be working

--- a/dashboard/Dockerfile
+++ b/dashboard/Dockerfile
@@ -3,15 +3,11 @@ FROM node:13.12.0-alpine
 # set working directory
 WORKDIR /app
 
-# add `/app/node_modules/.bin` to $PATH
-ENV PATH /app/node_modules/.bin:$PATH
 ENV DIRECTORY dashboard
 
 # install app dependencies
-COPY $DIRECTORY/package.json $DIRECTORY/package-lock.json ./
-#RUN npm install
+COPY $DIRECTORY/package*.json ./
 RUN npm clean-install
-#RUN npm install react-scripts -g
 
 # add app
 COPY ./$DIRECTORY ./

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -90,7 +90,7 @@ services:
       - REACT_APP_EXECUTION_API_HOST=http://localhost:5002
     volumes:
       - ./dashboard:/app
-
+      - /app/node_modules
 volumes:
   postgres:
     driver: local

